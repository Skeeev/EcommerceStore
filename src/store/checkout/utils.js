import get from 'lodash/get';
import find from 'lodash/find';

const normalizeProducts = products =>
  products.map(({ key, name: title, price, total, quantity, thumb }) => ({
    id: `${key}`,
    title,
    price,
    total,
    quantity,
    imageUri: `http:${thumb}`
  }));

export const extractCheckoutData = ({ products, totals }) => ({
  products: normalizeProducts(products),
  total: get(find(totals, { id: 'total' }), 'text', '$00.00'),
  totalQuantity: products.reduce((total, { quantity }) => total + quantity, 0)
});
