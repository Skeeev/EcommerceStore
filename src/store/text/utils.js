export const prepareTextPayload = text => ({
  payload: {
    text
  }
});
