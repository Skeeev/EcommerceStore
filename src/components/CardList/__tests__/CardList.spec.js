import React from 'react';
import { shallow } from 'enzyme';

import { getComponentDisplayName } from 'src/utils/displayName';
import { CardList } from '..';

describe(`${getComponentDisplayName(CardList)} component`, () => {
  const renderComponent = props => shallow(<CardList {...props} />);

  it('renders correctly', () => {
    const component = renderComponent({
      data: [
        {
          id: 'dummy-id',
          itemProp: 'dummy prop'
        }
      ]
    });

    expect(component).toMatchSnapshot();
  });
});
