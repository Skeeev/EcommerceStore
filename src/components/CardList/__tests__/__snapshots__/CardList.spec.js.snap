// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList component renders correctly 1`] = `
<View>
  <FlatList
    data={
      Array [
        Object {
          "id": "dummy-id",
          "itemProp": "dummy prop",
        },
      ]
    }
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    refreshControl={null}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</View>
`;
